src/components/InventoryGroups/Modals/RemoveHostsFromGroupModal.js

```diff
@@ -7,62 +7,78 @@ import { removeHostsFromGroup } from '../utils/api';
 import componentTypes from '@data-driven-forms/react-form-renderer/component-types';
 import { Text } from '@patternfly/react-core';
 
-const schema = (groupName, hosts) => ({
-  fields: [
-    {
-      component: componentTypes.PLAIN_TEXT,
-      name: 'warning-message',
-      label:
-        hosts.length === 1 ? (
-          <Text>
-            <strong>{hosts[0].display_name}</strong> will no longer be part of{' '}
-            <strong>{groupName}</strong> and its configuration will be impacted.
-          </Text>
-        ) : (
-          <Text>
-            <strong>{hosts.length}</strong> systems will no longer be part of{' '}
-            <strong>{groupName}</strong> and their configuration will be
-            impacted.
-          </Text>
-        ),
-    },
-  ],
-});
+const schema = (hosts) => {
+  const hostsInGroup = hosts.filter(({ groups }) => groups.length > 0); // selection can contain ungroupped hosts
+  const groupName = hostsInGroup[0].groups[0].name;
+
+  return {
+    fields: [
+      {
+        component: componentTypes.PLAIN_TEXT,
+        name: 'warning-message',
+        label:
+          hostsInGroup.length === 1 ? (
+            <Text>
+              <strong>{hostsInGroup[0].display_name}</strong> will no longer be
+              part of <strong>{groupName}</strong> and its configuration will be
+              impacted.
+            </Text>
+          ) : (
+            <Text>
+              <strong>{hostsInGroup.length}</strong> systems will no longer be
+              part of <strong>{groupName}</strong> and their configuration will
+              be impacted.
+            </Text>
+          ),
+      },
+    ],
+  };
+};
+
+const statusMessages = (hosts) => {
+  const hostsInGroup = hosts.filter(({ groups }) => groups.length > 0);
+  const groupName = hostsInGroup[0].groups[0].name;
+
+  return hostsInGroup.length === 1
+    ? {
+        onSuccess: {
+          title: `1 system removed from ${groupName}`,
+        },
+        onError: {
+          title: `Failed to remove 1 system from ${groupName}`,
+        },
+      }
+    : {
+        onSuccess: {
+          title: `${hostsInGroup.length} systems removed from ${groupName}`,
+        },
+        onError: {
+          title: `Failed to remove ${hostsInGroup.length} systems from ${groupName}`,
+        },
+      };
+};
 
 const RemoveHostsFromGroupModal = ({
   isModalOpen,
   setIsModalOpen,
   modalState: hosts,
   reloadData,
 }) => {
-  const dispatch = useDispatch();
-  // the current iteration of groups feature a host can be in at maximum one group
-  const { name: groupName, id: groupId } = hosts[0].groups[0];
+  // TODO: should soon support removal from more than one group at once
```

Comment:
Do you intend to keep this comment ?
If not, this is just a reminder to remove it :)

Reply:
Yeah, it doesn't have any special call for action, but I rather wanted to keep it as helping tip. But I better remove it completely